# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/device/gl.rs
# Commit: 86db847cfe57
# Full Hash: 86db847cfe570dba25289ffbe52aaaaecf76bc70
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-18 04:07:40
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r c3495de132cd -r 86db847cfe57 gfx/wr/webrender/src/device/gl.rs
--- a/gfx/wr/webrender/src/device/gl.rs	Thu Dec 17 19:53:54 2020 +0000
+++ b/gfx/wr/webrender/src/device/gl.rs	Thu Dec 17 19:54:02 2020 +0000
@@ -475,6 +475,13 @@
         self.filter
     }
 
+    pub fn is_array(&self) -> bool {
+        match self.target {
+            gl::TEXTURE_2D_ARRAY => true,
+            _ => false
+        }
+    }
+
     pub fn supports_depth(&self) -> bool {
         !self.fbos_with_depth.is_empty()
     }